### PR TITLE
fix: add catergories in a seperate property field in the property bag

### PR DESCRIPTION
**Pull request by Spoor2709**
- [ ] Ready for review

#### What does this PR do?
This PR removes the possibility of having a tag break the category filtering by giving categories it own place in the property bag

This also changes the analysisResults.files[filePath] test to a snapshot due to constant updating of the fingerprinting

### CODE REVIEW

**Review by orkamara — CHANGES_REQUESTED** [review]
Still not sure about this change.... it means we don't follow Sarif format anymore.... http://docs.oasis-open.org/sarif/sarif/v2.0/csprd01/sarif-v2.0-csprd01.html#_Toc517435925

File: src/sarif_converter.ts [hunk 1]
```diff
@@ -67,10 +70,11 @@ function getRulesAndAllIssues(analysisResults: IAnalysisResult, allIssuesBySugge
     const language = suggestion.id.split('%2F')[0];
     const suggestionId = `${language}/${suggestion.rule}`;
     const ruleProperties: RuleProperties = {
-      tags: [language, ...suggestion.tags, ...suggestion.categories],
+      tags: [language, ...suggestion.tags],
+      categories: suggestion.categories,
```
Comment (Spoor2709) [inline src/sarif_converter.ts:74]: @ork what about:
```suggestion
      tags: [language, ...suggestion.tags, ...categories],
      categories: suggestion.categories,
```

Comment (orkamara) [reply src/sarif_converter.ts:74]: I mean, `categories` is not part of Sarif....

Comment (Spoor2709) [reply src/sarif_converter.ts:74]: yes but nor is exampleCommitFixes, exampleCommitDescriptions, precision.

**Review by orkamara — APPROVED** [review]

**Pull request merged**